noir/noir-repo/compiler/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -217,20 +217,37 @@ impl CrateDefMap {
                         })
                         .collect();
 
-                    let events = module
-                        .type_definitions()
-                        .filter_map(|id| {
-                            id.as_type().filter(|struct_id| {
-                                interner
-                                    .struct_attributes(struct_id)
-                                    .iter()
-                                    .any(|attr| attr == &SecondaryAttribute::Event)
-                            })
-                        })
-                        .collect();
+                    let mut outputs =
+                        ContractOutputs { structs: HashMap::new(), globals: HashMap::new() };
+
+                    interner.get_all_globals().iter().for_each(|global_info| {
```

Comment:
I think this is still pulling in all globals from dependencies.

Reply:
I put the block at the resolver level, which would cause a compilation error. Is it really necessary to prevent it at this level?